extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.DesktopArticleTarget.js

```diff
@@ -961,20 +962,28 @@ ve.init.mw.DesktopArticleTarget.prototype.restoreScrollPosition = function () {
 /**
  * Hide the toolbar.
  *
+ * @param {boolean} [noAnimate] Don't animate toolbar teardown
  * @return {jQuery.Promise} Promise which resolves when toolbar is hidden
  */
-ve.init.mw.DesktopArticleTarget.prototype.teardownToolbar = function () {
+ve.init.mw.DesktopArticleTarget.prototype.teardownToolbar = function ( noAnimate ) {
 	var target = this,
-		deferred = $.Deferred();
-	this.toolbar.$element.css( 'height', this.toolbar.$bar.outerHeight() );
-	setTimeout( function () {
-		target.toolbar.$element.css( 'height', '0' );
-		target.toolbar.$element.one( 'transitionend', function () {
-			target.toolbar.destroy();
-			target.toolbar = null;
-			deferred.resolve();
+		deferred = $.Deferred(),
+		tearDownToolbar = function () {
+		target.toolbar.destroy();
```

Comment:
Something off with the indentations. Also, when it comes to more complex constructs, like functions, we tend to intialize them separately - instead of doing it with all the other vars - seems to make code a bit more readable.